corehq/apps/app_manager/models.py

```diff
@@ -5317,11 +5354,12 @@ def get_locations(cls, suite):
     def SUITE_XPATH(self):
         return 'suite/resource/location[@authority="local"]'
 
-    def create_all_files(self):
+    def create_all_files(self, build_profile=None):
         files = {
             'profile.xml': self.create_profile(),
         }
         tree = _parse_xml(files['profile.xml'])
+        langs_for_build = self.langs if not build_profile else self.build_profiles[build_profile].langs
```

Comment:
kinda feels like this could be a function on the app rather than having to repeat the same logic everywhere